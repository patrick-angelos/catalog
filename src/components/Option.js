import React from 'react';
import PropTypes from 'prop-types';

const Option = (props) => {
  const { id, options } = props;
  return (
    <div className="option-wrapper flex space-between relative">
      <select className="option" id={id}>{options}</select>
      <div className="arrow absolute" />
    </div>
  );
};

Option.propTypes = {
  options: PropTypes.arrayOf(PropTypes.element.isRequired).isRequired,
  id: PropTypes.string.isRequired,
};

export default Option;
