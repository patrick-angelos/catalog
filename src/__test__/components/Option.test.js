import React from 'react';
import renderer from 'react-test-renderer';
import Option from '../../components/Option';

describe('Option', () => {
  it('renders correctly', () => {
    const options = ['1', '2', '3'].map((option) => <option key={option}>{option}</option>);
    const tree = renderer
      .create(<Option
        id="cuisineType"
        options={options}
        value="2"
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
