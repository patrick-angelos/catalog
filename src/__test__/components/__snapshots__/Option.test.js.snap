// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option renders correctly 1`] = `
<div
  className="option-wrapper flex space-between relative background-white"
>
  <select
    className="background-white option"
    defaultValue="2"
    id="cuisineType"
  >
    <option>
      1
    </option>
    <option>
      2
    </option>
    <option>
      3
    </option>
  </select>
  <div
    className="arrow absolute"
  />
</div>
`;
